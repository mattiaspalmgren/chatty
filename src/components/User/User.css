.User {
    composes: padding-medium from "../../utils/utils.css";
    border: 1px solid gainsboro;
    width: 30%;
}