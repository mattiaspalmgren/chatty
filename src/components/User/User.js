import React from "react";

import Button from "../Button/Button.js";
import style from "./User.css";

const User = ({ user }) => (
    <div className={style.User}>
        <h2>{user.name}</h2>
        <p>{user.text}</p>
        <Button/>
    </div>
);

export default User;