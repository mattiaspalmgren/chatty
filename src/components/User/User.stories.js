import React from 'react';
import { storiesOf } from '@storybook/react';
import { actions } from '@storybook/addon-actions';

import User from './User.js';

const eventsFromObject = actions({ onMouseOver: 'hovered' });

const user = {
    name: "Alice",
    text: "Lorem ipsum dolor sit amet, consectetur adipiscing elit." +
    " Donec ligula arcu, mollis at consectetur sit amet, porta gravida" +
    " ligula. Aenean pellentesque lobortis urna."
};

storiesOf('User', module)
  .add('with text', () => <User user={user}></User>)
