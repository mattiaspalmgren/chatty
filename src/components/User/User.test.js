import React from 'react';
import { shallow } from 'enzyme';
import User from './User.js';

it('render user shallow correctly', () => {
  const user = {
    name: "Alice",
    text: "Lorem ipsum dolor sit amet, consectetur adipiscing elit." +
    " Donec ligula arcu, mollis at consectetur sit amet, porta gravida" +
    " ligula. Aenean pellentesque lobortis urna."
  };
  const component = shallow(<User user={user} />);
  expect(component).toMatchSnapshot();
});