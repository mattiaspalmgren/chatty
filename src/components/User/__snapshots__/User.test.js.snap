// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render user shallow correctly 1`] = `
<div
  className="User"
>
  <h2>
    Alice
  </h2>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec ligula arcu, mollis at consectetur sit amet, porta gravida ligula. Aenean pellentesque lobortis urna.
  </p>
  <Button />
</div>
`;
